tests/misc_tests.c

```diff
@@ -463,6 +463,19 @@ static void cjson_get_string_value_should_get_a_string(void)
     cJSON_Delete(string);
 }
 
+static void cjson_get_number_value_should_get_a_number(void)
+{
+    cJSON *string = cJSON_CreateString("test");
+    cJSON *number = cJSON_CreateNumber(1);
+
+    TEST_ASSERT_TRUE(cJSON_GetNumberValue(number) == number->valuedouble);
+    TEST_ASSERT_TRUE(is_nan(cJSON_GetNumberValue(string)));
+    TEST_ASSERT_TRUE(is_nan(cJSON_GetNumberValue(NULL)));
```

Comment:
firstly, using `==` to compare two float number is not a good way;
secondly, `is_nan` has not been defined. if you mean isnan(in math.h), then it is not applied in c89.
Unity test framwork has plenty of asserts can help you do this.
```suggestion
    TEST_ASSERT_EQUAL_DOUBLE(cJSON_GetNumberValue(number), number->valuedouble);
    TEST_ASSERT_DOUBLE_IS_NAN(cJSON_GetNumberValue(string));
    TEST_ASSERT_DOUBLE_IS_NAN(cJSON_GetNumberValue(NULL));
```

Reply:
You already fix it. Thank you for review !